tests/test_dem.py

```diff
@@ -22,6 +23,35 @@ def test_load(self):
         fn_img = xdem.examples.FILEPATHS["longyearbyen_ref_dem"]
         img = DEM(fn_img)
 
+    def test_copy(self):
```

Comment:
A test should be added to assert the consistency of the `.data` attribute.

A quick version would be:
```python
dem1 = DEM("path/to/file")
dem1.data += 5

dem2 = dem1.copy()

assert np.mean(np.abs(dem1.data - dem2.data)) < 0.001
```
And I suspect the above test will fail right now.

Reply:
This test is done further down, the two data are consistent:
```python
# check data array
        assert np.count_nonzero(~img.data == img2.data) == 0
        # check dataset_mask array
        assert np.count_nonzero(~img2.dataset_mask() == img2.dataset_mask()) == 0
```

Reply:
But you haven't checked what happens if `img.data` is modified, right? If `img2` is loaded from `img.filename`, then it probably won't contain the in-memory modifications to `img.data`